forc-pkg/src/pkg.rs

```diff
@@ -1720,19 +1779,45 @@ pub fn sway_build_config(
 /// then the std prelude will also be added.
 pub fn dependency_namespace(
     namespace_map: &HashMap<NodeIx, namespace::Module>,
+    compiled_map: &HashMap<NodeIx, Compiled>,
     graph: &Graph,
     node: NodeIx,
     constants: BTreeMap<String, ConfigTimeConstant>,
 ) -> Result<namespace::Module, vec1::Vec1<CompileError>> {
-    let mut namespace = namespace::Module::default_with_constants(constants)?;
+    let mut namespace = namespace::Module::default_with_constants(constants.clone())?;
 
     // Add direct dependencies.
     let mut core_added = false;
     for edge in graph.edges_directed(node, Direction::Outgoing) {
         let dep_node = edge.target();
-        let dep_namespace = &namespace_map[&dep_node];
-        let dep_name = kebab_to_snake_case(edge.weight());
-        namespace.insert_submodule(dep_name, dep_namespace.clone());
+        let dep_name = kebab_to_snake_case(&edge.weight().name);
+        let dep_edge = edge.weight();
+        let dep_namespace = match dep_edge.kind {
+            DepKind::Library => namespace_map
+                .get(&dep_node)
+                .map(Clone::clone)
+                .expect("no namespace module for {dep_name}"),
+            DepKind::Contract => {
+                let mut constants = constants.clone();
```

Comment:
Yeah it seems like it is not doing what i intended at the beginning. So if a contract dep declares config time constants that are `public` they are not transitively moved to root node. Since the infrastructure that we are using is not going to be used for config time constants, but if I understood correctly, I guess it does not need to work as this infra is going to be just used our `CONTRACT_ID` logic and config time constants are going to be separate.